net/graphsync_fetcher.go

```diff
@@ -55,23 +55,41 @@ type graphsyncFallbackPeerTracker interface {
 // GraphSyncFetcher is used to fetch data over the network.  It is implemented
 // using a Graphsync exchange to fetch tipsets recursively
 type GraphSyncFetcher struct {
-	exchange    GraphExchange
-	validator   consensus.SyntaxValidator
-	store       bstore.Blockstore
-	ssb         selectorbuilder.SelectorSpecBuilder
-	peerTracker graphsyncFallbackPeerTracker
+	exchange            GraphExchange
+	validator           consensus.SyntaxValidator
+	store               bstore.Blockstore
+	ssb                 selectorbuilder.SelectorSpecBuilder
+	peerTracker         graphsyncFallbackPeerTracker
+	unresponsiveTimeout time.Duration
+}
+
+// GraphsyncFetcherOption is function that configures graphsync. It should not
+// be created directly but should instead generated through an option function like
+// UseFetcherTimeout
+type GraphsyncFetcherOption func(*GraphSyncFetcher)
+
+// UseUnresponsiveTimeout sets up the GraphsyncFetcher with a different
+// unresponsiveness timeout than the default
+func UseUnresponsiveTimeout(timeout time.Duration) GraphsyncFetcherOption {
+	return func(gsf *GraphSyncFetcher) {
+		gsf.unresponsiveTimeout = timeout
+	}
 }
 
 // NewGraphSyncFetcher returns a GraphsyncFetcher wired up to the input Graphsync exchange and
 // attached local blockservice for reloading blocks in memory once they are returned
 func NewGraphSyncFetcher(ctx context.Context, exchange GraphExchange, blockstore bstore.Blockstore,
-	bv consensus.SyntaxValidator, pt graphsyncFallbackPeerTracker) *GraphSyncFetcher {
+	bv consensus.SyntaxValidator, pt graphsyncFallbackPeerTracker, options ...GraphsyncFetcherOption) *GraphSyncFetcher {
```

Comment:
It seems off to use the function options pattern for setting the timeout dependency while setting the other dependencies with explicit inputs.  I could be missing the differentiating factor though.